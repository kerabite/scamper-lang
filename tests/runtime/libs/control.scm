(define inc
  (lambda (x) (+ x 1)))

(define l (list 1 2 3 4 5 6 7 8 9 10))

(map inc l)

(map (lambda (p) (car p))
  (list (cons "a" "b") (cons "c" "d") (cons "e" "f")))

(map inc null)

(map - (list 3 5 7 1 9)
       (list 1 2 3 0 13))

(map (lambda (x y z) (if x y z))
     (list #t #f #t)
     (list "yes" "no" "maybe")
     (list "y" "n" "m"))

(filter
  (lambda (v)
    (= (remainder v 2) 0))
  l)

(filter
  (lambda (v)
    (>= v 5))
  l)

(filter
  (lambda (v)
    (< v 5))
  l)

(filter (lambda (v) (> v 20)) l)

(filter (lambda (v) #t) null)

(fold + 0 l)
(reduce + l)