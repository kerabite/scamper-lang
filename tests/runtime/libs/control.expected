(list 2 3 4 5 6 7 8 9 10 11)
(list "a" "c" "e")
null
(list 2 4 6 8 10)
(list 5 6 7 8 9 10)
(list 1 2 3 4)
null
null
55
55
